content/go/concepts/interfaces/interfaces.md

```diff
@@ -0,0 +1,268 @@
+---
+Title: 'Interfaces'
+Description: 'An interface is a set of method signatures.'
+Subjects:
+  - 'Code Foundations'
+  - 'Computer Science'
+Tags:
+  - 'Control Flow'
+  - 'Interfaces'
+CatalogContent:
+  - 'learn-go'
+  - 'paths/back-end-engineer-career-path'
+---
+
+An interface is composed of set of method signatures. These method signatures define the input and return values of which a data type or struct can conform to. In order to implement an interface, the type or struct must satisfy all the defined method signatures.
+
+## Creating an Interface
+
+```go
+type InterfaceName interface {
+    method() return_type
+}
+```
+
+A type implements an interface by implementing its methods, In `Go`, interfaces are implemented implicitly and enables polymorphism.
+
+### Example
+
+```go
+package main
+
+import (
+  "fmt"
+)
+
+type marriage interface {
+  love()
+  arranged()
+}
+
+type john struct{}
+
+func (a *john) love() {
+  fmt.Println("love with john")
+}
+
+func (a *john) arranged() {
+  fmt.Println("arranged marriage with john")
+}
+
+type max struct{}
+
+func (a *max) love() {
+  fmt.Println("love with max")
+}
+
+func (a *max) arranged() {
+  fmt.Println("arranged marriage with max")
+}
+
+func newLife(guy marriage) {
+  guy.love()
+  guy.arranged()
+}
+
+func main() {
+  newLife(&john{})
+  newLife(&max{})
+}
+```
+
+Output:
+
+```shell
+love with john
+arranged marriage with john
+love with max
+arranged marriage with max
+```
+
+In the above example:
```

Comment:
@yangc95  if you meant explaination to above example its given here.... heading is it just comes under creating an interface hm n yeah it makes more sense to move line abt polymorphism down will do tht

Reply:
@Pavan-04-08 This is more for pedagogical reasons. Creating an interface can be broken down into how to create an interface (with the interface keyword) and then how to implement an interface (wether that be done using a type or structure)

It would be good to show just one example first and keep only one `struct` (John or max) to better display this. Don't forget to adjust the example below to match!

If you disagree or have concerns, I'd love to discuss more

Reply:
@yangc95  ok... i got what you trying to say 
i would like to suggest something like https://www.codecademy.com/resources/docs/java/array-list
if we are going to keep things simpler doing like above like say array-list methods are not written in same page as rraylist definition 

likewise we can make interfaces too

on interfaces.md ---->interfaces explaintion with single struct n two structs
and then similar to arraylist methods at down on page
we can do for interfaces like say
empty interface
embedding interface with structs
interface conversion...etc

Reply:
in this way we can be more details about things ig even if we dont add it would be like not overwhelming for a newbie who is lookin at docs here (in any case) as he go step by step instead of seeing a chunk abt interfaces at one place

Reply:
@Pavan-04-08 I think another issue here is that repeating the same examples (but with different names e.g., John and max) don't provide additional value or any new information here.

I can assist you with this change as well, if you give me a thumbs up 👍 

In addition, I noticed that you have many commits on this PR. If this is cumbersome, there's an option to add it to a batch, allowing you to push everything up in one commit. Hope this helps